### PR TITLE
Add Module Federation Support

### PR DESCRIPTION
**Pull request by hasanayan**
TL;DR;

the plugin now injects additional SENTRY_RELEASES variable into global scope as an object that holds versions for all apps keyed by [project]@[org]

-----
When ModuleFederationPlugin is used, assuming the project has a single entry in the webpack config, the build outputs a bundle with two separate entrypoints; usually named as 'main' and 'remoteEntry'. Main is used when the app is running in standalone (host) mode and remoteEntry is used when it is consumed as a remote by another app. In our projects we needed to access the version value for all remotes on our host app so that we can create SentryClient per remote with their own respective versions.

With this PR, the code injected by loader is improved to include a second variable `SENTRY_RELEASES` in addition to `SENTRY_RELEASE`. This code goes to main module and is not loaded when the app is running as remote. So I also added the part to tap into compilation and inject the variable in remoteEntry as well.

If this is accepted, we can maybe open another PR on @sentry/browser so that the default release is read from SENTRY_RELEASES and retire SENTRY_RELEASE

### CODE REVIEW

**Comment by hasanayan** [comment]
@kamilogorek  @AbhiPrasad this PR is ready for review now. Is there anything I can do to help you review this?

**Comment by kamilogorek** [comment]
Thanks, I'll try to review it this week

**Comment by kamilogorek** [comment]
Code looks mostly fine, however integration tests are failing right now.

(I'll be gone for the next 2 weeks, vacations time)

**Comment by hasanayan** [comment]
@kamilogorek it's been a while since I looked at the code, can I run these tests locally? Otherwise, could you grant me permission to run the tests whenever I push code to this branch?

**Comment by kamilogorek** [comment]
https://github.com/getsentry/sentry-webpack-plugin/blob/master/.github/workflows/ci.yml

```
npm run test
npm run test:integration
```

These are only test commands run in the CI.

**Comment by hasanayan** [comment]
@kamilogorek alright, I'll fix the problems in the coming days. Happy vacations.

**Comment by hasanayan** [comment]
re-created this PR in https://github.com/getsentry/sentry-webpack-plugin/pull/307
